src/move_basic.cpp

```diff
@@ -725,6 +736,25 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
         remaining = goalInBase.getOrigin();
         double distRemaining = sqrt(linear.x() * linear.x() + linear.y() * linear.y());
 
+// 
+	if (distRemaining < prevDistance) {
+		double current = ros::Time::now().toSec();
+		if (current-last > abortTimeout) {
+			abortGoal("MoveBasic: Goal update timeout.");		
+		}
+		prevDistance = distRemaining;
+		last = current;
+	}
+
+	if (distRemaining < prevDistance) {
+		if (prevDistance+distThreshold < distRemaining) {
+			abortGoal("MoveBasic: Goal distance threshold exceeded.");		
```

Comment:
Same thing here:

"Detected that we are moving further from the goal, aborting."